src/sdk/namespace_s3.js

```diff
@@ -228,11 +245,27 @@ class NamespaceS3 {
 
             this._assign_encryption_to_request(params, request);
 
-            res = await this.s3.putObject(request).promise();
+            res = await this.s3.putObject(request).on('error', err => {
+                this._translate_error_code(err);
+                dbg.warn('NamespaceS3.upload_object:', inspect(err));
+                return err;
+            }).on('success', response => {
+                dbg.log0('NamespaceS3.upload_object:',
+                    'status code', response.httpResponse.statusCode,
+                    'etag', response.httpResponse.headers.etag,
+                    'versionId', response.httpResponse.headers.VersionId,
+                    'created date', response.httpResponse.headers.date
+                );
+
+                if (response.httpResponse.statusCode > 200) return; // will be handled by error event
+                last_modify_date = response.httpResponse.headers.date;
+                statusCode = response.httpResponse.statusCode;
+                return response;
+            }).promise();
         }
         dbg.log0('NamespaceS3.upload_object:', this.bucket, inspect(params), 'res', inspect(res));
         const etag = s3_utils.parse_etag(res.ETag);
-        return { etag, version_id: res.VersionId };
+        return { etag, statusCode: statusCode, version_id: res.VersionId, date: last_modify_date };
```

Comment:
I was using the statusCode to check if the upload operation succeeded. Instead of using `statusCode`, I just check if the date is set in the response in namespace_cache.js